### DISCUSSION TITLE
rotate

### DISCUSSION THREAD

**Discussion by Sarabanga** [savonet/liquidsoap#2191] [General]
There is almost no information about this in the book.
I would like to know how to do 
Two blocks of 1 jingle & 2 music, then 1 jingle,1 promo,
```rotate (weights = [1, 2, 1, 2, 1, 1], [jingle, music, jingle, music, jingle, promo]) ``` does not work... Only one file is playing from music. And code looks like a scary crocodile 

Ok. Go another way
```
block = rotate (weights = [1, 2], [jingle, music])
air = rotate (weights = [6, 1, 1], [block, jingle, promo])
```
Six files from block then jingle and promo.
Hm... No way, dude. Not today.
Сomplete chaos of jingles, music and promos

Help!

**Comment by toots** [comment]
Hi,

Your code looks correct. The `rotate` operator was (again) cleaned up in `v2.0.3-preview`. Any chance you could try this version? You should have some builds of it here: https://github.com/savonet/liquidsoap/actions/runs/1775654421 or else you can use the `opam pin` mechanism:
```
git clone https://github.com/savonet/liquidsoap.git
cd liquidsoap && git checkout v2.0.3-preview
opam install -y .
```

**Comment by Sarabanga** [comment]
2.0.3 preview
```
rotate (weights = [1, 2, 1, 2, 1, 1], [jingle, music, jingle, music, jingle, promo])
``` 
OK

```
block = rotate (weights = [1, 2], [jingle, music])
air = rotate (weights = [6, 1, 1], [block, jingle, promo]) 
```  
plays only block part.
